tests/testrides.py

```diff
@@ -0,0 +1,121 @@
+import json
+import unittest
+
+from application import create_app
+
+
+class RidesofferTests(unittest.TestCase):
+
+    def setUp(self):
+        """Prepare testing environment """
+
+        self.app = create_app('testing')
+        self.app = self.app.test_client()
+
+    def tearDown(self):
+        """Release flask app instance"""
+        self.app = None
+
+    def test_create_ride(self):
+        """test user can create a ride"""
+        ride = {
+            "start point": "Witeithye",
+            "destination": "Ngara",
+            "route": "Thika superhighway",
+            "start time": "June 10 2018 6:00AM",
+            "available space": 10
+        }
+        response = self.app.post('/api/v1/rides',
+                                 data=json.dumps(ride),
+                                 content_type='application/json')
+        self.assertEqual(response.status_code, 201)
+        response_data = json.loads(response.get_data().decode('utf-8'))
+        self.assertEqual(response_data['message'],
+                         'ride offer added successfully.')
+
+    def test_cannot_create_ride_with_wrong_date_time(self):
+        """Tests that date is parsed in the specified format
+        The date format is Month day Year hour:minutes"""
+        ride = {
+            "start point": "Witeithye",
+            "destination": "Ngara",
+            "route": "Thika superhighway",
+            "start time": "111111",
+            "available space": 10
+        }
+        response = self.app.post('/api/v1/rides',
+                                 data=json.dumps(ride),
+                                 content_type='application/json')
+        self.assertEqual(response.status_code, 400)
+        response_data = json.loads(response.get_data().decode('utf-8'))
+        self.assertEqual(response_data['message'],
+                         "use correct format for date and time.")
+
+    def test_cannot_create_ride_without_details(self):
+        ride = {}
+        response = self.app.post('/api/v1/rides',
+                                 data=json.dumps(ride),
+                                 content_type='application/json')
+        self.assertEqual(response.status_code, 400)
+        response_data = json.loads(response.get_data().decode('utf-8'))
+        self.assertEqual(response_data['message'],
+                         "make sure you provide all required fields.")
+
+    def test_available_space_can_only_be_numbers(self):
+        ride = {
+            "start point": "Witeithye",
+            "destination": "Ngara",
+            "route": "Thika superhighway",
+            "start time": "June 10 2018 6:00AM",
+            "available space": "10"
```

Comment:
Since the string **"10"** can be converted to a number, try testing with letters or other characters

Reply:
I have modified it to use 'Ten' instead of "10"